src/puppetlabs/puppetdb/scf/storage_utils.clj

```diff
@@ -35,6 +35,34 @@
 
 ;; FUNCTIONS
 
+(defn deregister-unused-drivers
+  "Given a database config or connection pool, deregister the driver
+   (hsql or postgres) that isn't being used, and return the initial driver
+   list for reregistration in case it's required."
+  [db-config]
+  (let [configured-driver (java.sql.DriverManager/getDriver
+                            (if-let [datasource (:datasource db-config)]
+                              (.getJdbcUrl datasource)
+                              (format "jdbc:%s:%s"
+                                      (:subprotocol db-config)
+                                      (:subname db-config))))
+        initial-drivers (enumeration-seq (java.sql.DriverManager/getDrivers))]
+    (doseq [driver initial-drivers
+            :when (not= (class driver) (class configured-driver))]
+      (java.sql.DriverManager/deregisterDriver driver))
+    initial-drivers))
```

Comment:
I understand why your return `initial-drivers` here after reading through the test fixtures, but it's not really intuitive in terms of what this function does. I would expect this function to return all the drivers that are still registered after deregistering. It seems like anything that needs to keep around `initial-drivers` could do so through a `get-drivers` function.